TaintingChecker false-positive for generics

Example:

```
public class Test {

  private final @Untainted Set<@Untainted String> s;

  public Test() {
    s = new @Untainted TreeSet<@Untainted String>();
  }
}
```

is correct (with Java 7 warning for redundant type specification), while

```
public class Test {

  private final @Untainted Set<@Untainted String> s;

  public Test() {
    s = new @Untainted TreeSet<>();
  }
}
```

emits `assignment.type.incompatible`, the TaintingChecker thinks that the `TreeSet` is `@Tainted`.

Note, that `new @Untainted TreeSet</*@Untainted String*/>();` also emits the false-positive, and it's also present if the generic type parameter does not have any annotations.

The Tainting Checker issues this warning (for the code snippet in the first comment):
```
TaintingTest.java:9: error: [assignment.type.incompatible] incompatible types in assignment.
    s = new @Untainted TreeSet<>();
        ^
  found   : @TAINTED TreeSet<@UNTAINTED String>
  required: @UNTAINTED Set<@UNTAINTED String>
1 error
```

--- a/checker/tests/tainting/TaintingDiamondInference.java
+++ b/checker/tests/tainting/TaintingDiamondInference.java
@@ -1,8 +1,5 @@
 // Test case for issue #660: https://github.com/typetools/checker-framework/issues/660
 
-// The test is skipped until the issue is fixed:
-// @skip-test
-
 import java.util.Set;
 import java.util.TreeSet;
 import org.checkerframework.checker.tainting.qual.Untainted;

--- a/checker/tests/tainting/TaintingDiamondInference.java
+++ b/checker/tests/tainting/TaintingDiamondInference.java
@@ -9,9 +9,9 @@ public class TaintingDiamondInference {
     private @Untainted Set<@Untainted String> s;
 
     public TaintingDiamondInference() {
-        // :: error: (cast.unsafe.constructor.invocation)
+        // :: warning: (cast.unsafe.constructor.invocation)
         s = new @Untainted TreeSet<>();
-        // :: error: (cast.unsafe.constructor.invocation)
+        // :: warning: (cast.unsafe.constructor.invocation)
         s = new @Untainted TreeSet<@Untainted String>();
     }
 }

--- a/checker/tests/tainting/TaintingDiamondInference.java
+++ b/checker/tests/tainting/TaintingDiamondInference.java
@@ -9,7 +9,9 @@ public class TaintingDiamondInference {
     private @Untainted Set<@Untainted String> s;
 
     public TaintingDiamondInference() {
+        // :: error: (cast.unsafe.constructor.invocation)
         s = new @Untainted TreeSet<>();
+        // :: error: (cast.unsafe.constructor.invocation)
         s = new @Untainted TreeSet<@Untainted String>();
     }
 }

